cEP-0024.md

```diff
@@ -0,0 +1,111 @@
+| Title    | Implement Gitmate automations and plugins for coala   |
+| Authors  | Vamshi Krishna Bommerla <mailto:bommvams72@gmail.com> |
+| Status   | Proposed                                              |
+| Type     | Process                                               |
+
+## Abstract
+
+This cEP describes the improvements to the coala workflow and the necessary
+automation to realise it, as a part of the
+[GSoC project](https://summerofcode.withgoogle.com/projects/#5344316227059712).
+
+## Introduction
+
+[GitMate](https://gitmate.io/) is an automation tool for developers. While it
+works well, there are a number of missing features which would save time of
+coala community, for example automatically close PRs opened on master of forks.
+Many of the plugins are general plugins, which are required by every
+organization, but coala needs more plugins which support coala development
+workflow. This project adds plugins to auto-review or auto-reject PRs,
+assign issues.
+
+## New Features
+
+### Auto-assign issue when it is mentioned in any of PR commits
+
+When a PR is opened in the repo and if any of its commit message
+mentions (Fixes, Closes) any unassigned issues and there is no CI failure, auto
+assign PR author to the mentioned unassigned issues. Comment on the
+PR (configurable) if someone else is working on the mentioned issue. If PR
+author is not a member of the organization, then he/she will be not be assigned
+to any of the issues.
+
+### Auto-reject PR opened on master of forks
+
+Normally, contributors should make changes in different branch from master
+branch of the forked repo, and submit a PR. But, sometimes coala receives PRs
+from master branch of the forked repo. To solve this problem, coala needs
+mechanism by which GitMate automatically closes those PRs and comments on the
+PR informing the PR author to work on a different branch. Comment message will
+be configurable.
+
+### Auto-mark PR with necessary labels as per the reviews done
+
+When a reviewer reviews a PR, he/she either approves changes or request changes
+or comments something when there is CI failure and leaves it there, but the PR
+would be still in the review queue. GitMate will automatically mark labels as
+per the following conditions:
+
+- WIP label conditions:
+  1. Reviewer requests changes
+  2. Anyone comments on the PR which has a CI failure
+- Approve label conditions:
+  1. Reviewer approves changes
+
+### Fixes vs Closes automated checking
+
+Newcomers often confuse when they should use `Fixes` or `Closes` in a commit
+message. So, GitMate should automatically check every commit for issues which
+are being Fixed or Closed, if those issues have bug label (configurable),
+`Fixes` should be used, else `Closes` should be used.
+When the commit message contains 'Fixes', but referenced issue doesn't have
+bug label, then user will be informed to request a maintainer that the issue
```

Comment:
what is "bug label"?

> then user will be informed ..

will this be a configurable message?   If yes, say it is configurable, and add a quoted block with the *exact* message that you believe coala should use for this configurable message so that we achieve the goals of this project.

I assume it will be a comment on the PR?